tests/lib/rules/jsx-filename-extension.js

```diff
@@ -74,6 +82,11 @@ ruleTester.run('jsx-filename-extension', rule, {
       filename: 'MyComponent.js',
       code: withJSXElement,
       errors: [{message: 'JSX not allowed in files with extension \'.js\''}]
+    }, {
+      filename: 'MyComponent.jsx',
+      code: withoutJSX,
+      options: [{allow: 'as-needed'}],
+      errors: [{message: 'Only files containing JSX may use the extension \'.jsx\''}]
```

Comment:
let's also add a test case for as-needed with `withJSXElement` and a `.js` file?

Reply:
Done